packages/langium/src/workspace/workspace-manager.ts

```diff
@@ -4,22 +4,26 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-import { CancellationToken } from 'vscode-languageserver';
+import { CancellationToken } from 'vscode-jsonrpc';
+import type { WorkspaceFolder } from 'vscode-languageserver-types';
```

Comment:
Question: Why import at all here, given that the `WorkspaceFolder` type is fairly trivial? IMO we shouldn't import from `vscode-languageserver-types` at all, even if it's just a `type` import.

Reply:
Personally, I'm always confused is several "copies" of the same type definition exist, and I'm always wondering which one to import. `WorkspaceFolder` is (transitively) available via `vscode-languageserver` and also via `vscode-languageclient`, which can bite you at runtime or during bundling, esp. if the developer isn't aware of that issue. That's why I think offering that interface via Langium and have stated and document the proper unique source is a good thing. However, if `vscode-languageserver-types` brings a too heavy overhead, we can revise that.

Reply:
> However, if vscode-languageserver-types brings a too heavy overhead, we can revise that.

It's not necessarily overhead but maintenance effort. We have to ensure on every change that we don't accidentally change this import from a `type` import to a real import.

Reply:
Would it be an option to import the interfaces and symbols (like `CancellationToken.None`) via one or tow TS modules, pre-bundle them as part of `npm run build`, and than rely on those pre-processed definitions?
Than we wouldn't copy-paste-clones of definitions from `vscode-languageserver....` but (ideally) have full control over the set of types and symbols being used.